crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4626,6 +4631,9 @@ impl<'db> TypeInferenceBuilder<'db> {
                 self.infer_type_expression(parameters);
                 todo_type!("generic type alias")
             }
+            KnownInstanceType::NoReturn | KnownInstanceType::Never => {
+                unreachable!("These types cannot be parametrized and have their own case")
+            }
```

Comment:
This means that we panic if a user writes something like this and then asks red-knot to check their code:

```py
import typing

x: typing.Never[int]
```

Backtrace:

```
thread '<unnamed>' panicked at crates/red_knot_python_semantic/src/types/infer.rs:4635:17:
internal error: entered unreachable code: These types cannot be parametrized and have their own case
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Rayon: detected unexpected panic; aborting
zsh: abort      cargo run -p red_knot -- --current-directory ../experiment
```

The annotation `typing.Never` is invalid but we shouldn't panic if the user writes something like this. Instead, we should emit a diagnostic and infer `Unknown`.

So it seems like this branch is pretty reachable ;)

Reply:
Thanks! I don't know why at the time I thought no branches go to this. Added `error: [invalid-type-parameter] "Type `typing.Never` expected no type parameter"` diagnostic and added a test for this case.

Reply:
Great, thank you!